Formula/openjdk.rb

```diff
@@ -123,38 +126,38 @@ def install
     ENV["MAKEFLAGS"] = "JOBS=#{ENV.make_jobs}"
     system "make", "images"
 
-    jdk = Dir["build/*/images/jdk-bundle/*"].first
-    libexec.install jdk => "openjdk.jdk"
-    bin.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/bin/*"]
-    include.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/include/*.h"]
-    include.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/include/darwin/*.h"]
+    on_macos do
+      jdk = Dir["build/*/images/jdk-bundle/*"].first
+      libexec.install jdk => "openjdk.jdk"
+      bin.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/bin/*"]
+      include.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/include/*.h"]
+      include.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/include/darwin/*.h"]
 
-    if Hardware::CPU.arm?
-      dest = libexec/"openjdk.jdk/Contents/Home/lib/JavaNativeFoundation.framework"
-      # Copy JavaNativeFoundation.framework from Xcode
-      # https://gist.github.com/claui/ea4248aa64d6a1b06c6d6ed80bc2d2b8#gistcomment-3539574
-      cp_r "#{framework_path}/JavaNativeFoundation.framework", dest, remove_destination: true
+      if Hardware::CPU.arm?
+        dest = libexec/"openjdk.jdk/Contents/Home/lib/JavaNativeFoundation.framework"
+        # Copy JavaNativeFoundation.framework from Xcode
+        # https://gist.github.com/claui/ea4248aa64d6a1b06c6d6ed80bc2d2b8#gistcomment-3539574
+        cp_r "#{framework_path}/JavaNativeFoundation.framework", dest, remove_destination: true
 
-      # Replace Apple signature by ad-hoc one (otherwise relocation will break it)
-      system "codesign", "-f", "-s", "-", "#{dest}/Versions/A/JavaNativeFoundation"
+        # Replace Apple signature by ad-hoc one (otherwise relocation will break it)
+        system "codesign", "-f", "-s", "-", "#{dest}/Versions/A/JavaNativeFoundation"
+      end
+    end
+
+    on_linux do
+      libexec.install Dir["build/linux-x86_64-server-release/images/jdk/*"]
+      bin.install_symlink Dir["#{libexec}/bin/*"]
+      include.install_symlink Dir["#{libexec}/include/*.h"]
+      include.install_symlink Dir["#{libexec}/include/linux/*.h"]
     end
   end
 
   def caveats
     on_macos do
-      s = <<~EOS
+      <<~EOS
         For the system Java wrappers to find this JDK, symlink it with
           sudo ln -sfn #{opt_libexec}/openjdk.jdk /Library/Java/JavaVirtualMachines/openjdk.jdk
       EOS
-
-      if Hardware::CPU.arm?
-        s += <<~EOS
-          This is a beta version of openjdk for Apple Silicon
-          (openjdk 16 preview).
```

Comment:
Probably should restore caveat so ARM mac users are aware that this is an Early-Access/Beta image and may experience more bugs than a GA release.